### PR TITLE
[dotnet new] telemetry for MSBuild evaluation

### PR DESCRIPTION
**Pull request by vlada-shubina**
- added telemetry for MSBuild evalution
- reversed behavior of project context evaluation: now it is available by default with option/env variable to disable it
- disabled telemetry for `dotnet new` tests

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs [hunk 1]
```diff
@@ -173,6 +186,27 @@ private MSBuildEvaluationResult EvaluateProjectInternal(IEngineEnvironmentSettin
...
+                    { "ProjectPath",  Sha256Hasher.HashWithNormalizedCasing(projectPath)},
+                    { "SdkStyleProject", IsSdkStyleProject.ToString() },
+                    { "TargetFrameworks", targetFrameworksString ?? "<null>"},
+                };
+                Dictionary<string, double> measurements = new()
+                {
+                    { "EvaluationTime",  watch.ElapsedMilliseconds },
+                    { "InnerEvaluationTime",  innerBuildWatch.ElapsedMilliseconds }
+                };
+                TelemetryEventEntry.TrackEvent("new/msbuild-eval", properties, measurements);
```
Comment (vlada-shubina) [inline src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs:208]: @baronfel could you please check this telemetry set?

File: src/Cli/dotnet/commands/dotnet-new/NewCommandParser.cs [hunk 2]
```diff
@@ -35,12 +35,12 @@ internal static class NewCommandParser
...
-        private const string EnableProjectContextEvaluationEnvVar = "DOTNET_CLI_ENABLE_PROJECT_EVAL";
+        private const string EnableProjectContextEvaluationEnvVar = "DOTNET_CLI_DISABLE_PROJECT_EVAL";
 
         private const string HostIdentifier = "dotnetcli";
         private static readonly Option<bool> _disableSdkTemplates = new Option<bool>("--debug:disable-sdk-templates", () => false, LocalizableStrings.DisableSdkTemplates_OptionDescription).Hide();
 
-        private static readonly Option<bool> _enableProjectContextEvaluation = new Option<bool>("--debug:enable-project-context", () => false, LocalizableStrings.EnableProjectContextEval_OptionDescription).Hide();
+        private static readonly Option<bool> _disableProjectContextEvaluation = new Option<bool>("--debug:disable-project-context", () => false, LocalizableStrings.DisableProjectContextEval_OptionDescription).Hide();
 
         internal static Option<FileInfo> ProjectPathOption { get; } = new Option<FileInfo>("--project", LocalizableStrings.ProjectPath_OptionDescription).ExistingOnly().Hide();
```
Comment (vlada-shubina) [inline src/Cli/dotnet/commands/dotnet-new/NewCommandParser.cs:45]: @baronfel should we make `--project` option visible?

File: src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs [hunk 3]
```diff
@@ -173,6 +186,27 @@ private MSBuildEvaluationResult EvaluateProjectInternal(IEngineEnvironmentSettin
...
                 return MSBuildEvaluationResult.CreateFailure(projectPath, e.Message);
             }
+            finally
+            {
+                watch.Stop();
+                if (innerBuildWatch.IsRunning)
+                {
+                    innerBuildWatch.Stop();
+                }
+                string? targetFrameworksString = targetFrameworks != null ? string.Join(",", targetFrameworks) : targetFramework;
```
Comment (baronfel) [inline src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs:196]: Can you hash each of these TFMs? the msbuild `targetframeworkeval` event hashes the TFM in it, and we should be consistent.

File: src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs [hunk 4]
```diff
@@ -173,6 +186,27 @@ private MSBuildEvaluationResult EvaluateProjectInternal(IEngineEnvironmentSettin
                 _logger?.LogDebug("Unexpected error: {0}", e);
                 return MSBuildEvaluationResult.CreateFailure(projectPath, e.Message);
             }
+            finally
+            {
+                watch.Stop();
+                if (innerBuildWatch.IsRunning)
```
Comment (JanKrivanek) [inline src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/MSBuildEvaluator.cs:192]: Condition can be removed - `Stopwatch.Stop` already checks for this.

**Review by JanKrivanek — APPROVED** [review]
LGTM

File: src/Cli/dotnet/commands/dotnet-new/NewCommandParser.cs (hunk 2 above)
Comment (baronfel) [reply src/Cli/dotnet/commands/dotnet-new/NewCommandParser.cs:45]: yes, for instanciate and list commands.

Comment (vlada-shubina) [reply src/Cli/dotnet/commands/dotnet-new/NewCommandParser.cs:45]: will move it in the next PR, together with output option refactoring.

**Pull request merged**